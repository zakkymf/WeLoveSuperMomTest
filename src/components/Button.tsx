import React from 'react';
import {Text, StyleSheet, TouchableOpacity} from 'react-native';
import {Colors} from '../themes';

interface ButtonProps {
  label: string;
  onPress: () => void;
}

const Button: React.FC<ButtonProps> = ({label, onPress}) => {
  return (
    <TouchableOpacity style={styles.button} onPress={onPress}>
      <Text style={styles.buttonText}>{label}</Text>
    </TouchableOpacity>
  );
};

const styles = StyleSheet.create({
  button: {
    height: 48,
    borderRadius: 8,
    marginVertical: 10,
    alignItems: 'center',
    justifyContent: 'center',
    backgroundColor: Colors.pink,
  },
  buttonText: {
    fontSize: 12,
    fontWeight: '700',
    color: Colors.white,
  },
});

export default Button;
