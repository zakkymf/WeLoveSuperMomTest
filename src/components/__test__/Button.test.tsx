import renderer from 'react-test-renderer';
import Button from '../Button';

describe('Button Component', () => {
  it('render button correctly', () => {
    const tree = renderer
      .create(<Button label="Export to CSV" onPress={() => jest.fn()} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
