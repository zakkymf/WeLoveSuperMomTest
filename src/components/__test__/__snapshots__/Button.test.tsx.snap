// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button Component render button correctly 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#E8157E",
      "borderRadius": 8,
      "height": 48,
      "justifyContent": "center",
      "marginVertical": 10,
      "opacity": 1,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#ffffff",
        "fontSize": 12,
        "fontWeight": "700",
      }
    }
  >
    Export to CSV
  </Text>
</View>
`;
